From 12fee3d24574bd084e8b6fa1dcaf84a084298c9a Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Fri, 13 Mar 2015 15:42:39 +0800
Subject: [PATCH] MLK-10410 net: fec: fix enet warning issue by suspend/resume

When down eth0 on i.MX6SX platform and then do suspend/resume
operation, there have one kernel dump. The patch fix the warning
issue. After eth interface down, it is not necessary to init phydev
to NULL since it do this in driver remove and eth open fucntion.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/net/ethernet/freescale/fec_main.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index 3db3470..47a81a1 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -2851,7 +2851,6 @@ fec_enet_close(struct net_device *ndev)
 	}
 
 	phy_disconnect(fep->phy_dev);
-	fep->phy_dev = NULL;
 
 	fec_enet_clk_enable(ndev, false);
 	pm_qos_remove_request(&ndev->pm_qos_req);
-- 
1.8.0

